extensions/extended-blocks/core-embed/facebook.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addFilter } from '@wordpress/hooks';
+import { registerBlockVariation, unregisterBlockVariation } from '@wordpress/blocks';
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getIconColor } from '../../shared/block-icons';
+
+const facebookVariation = {
+	// Deprecate Facebook Embed per FB policy
+	// See: https://developers.facebook.com/docs/plugins/oembed-legacy
+	name: 'facebook',
+	title: 'Facebook',
+	icon: {
+		src: 'facebook',
+		foreground: getIconColor(),
+	},
+	keywords: [ _x( 'social', 'block search term', 'jetpack' ) ],
+	description: __( 'Embed a Facebook post.', 'jetpack' ),
+	patterns: [ /^https?:\/\/www\.facebook.com\/.+/i ],
```

Comment:
I think `www` should be optional here, too.

Reply:
I copy-pasted the pattern from the [original GB embeds](https://github.com/WordPress/gutenberg/pull/24472/files#diff-d6e213880d8d5e45b0bf87afa293c01aL64) :thinking: Even though making the `www` optional makes sense to me, I'd rather not change the patterns TBH since I assume GB folks had them tried and tested :sweat_smile:

Reply:
Ok, what I'm trying to raise up if you paste a link without the `www`, the variation won't be applied and it seems to be an issue to me.

Reply:
Agree -- it _is_ surprising. I'm really just trying to be as consistent as possible with Core here :slightly_smiling_face: Note that they _did_ have the `www` marked optional for Instagram (so it seems like a conscious decision _not_ to have it for Facebook): https://github.com/WordPress/gutenberg/pull/24472/files#diff-d6e213880d8d5e45b0bf87afa293c01aL77

Maybe FB now has canonical redirects from `www.facebook.com` to `facbook.com`? Anyway, let's investigate later :+1: